test/cpu_marginrankingloss.hpp

```diff
@@ -0,0 +1,145 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#pragma once
+
+#include "tensor_holder.hpp"
+#include <miopen/tensor_view_utils.hpp>
+
+template <class T>
+void cpu_marginrankingloss_forward_5d(tensor<T> input1,
+                                      tensor<T> input2,
+                                      tensor<T> target,
+                                      tensor<T>& output,
+                                      float margin,
+                                      float divisor,
+                                      miopenMarginRakningLossReductionMode_t reduction_mode)
+{
+    tensor_view_t<5> I1_tv = get_inner_expanded_tv<5>(input1.desc);
+    tensor_view_t<5> I2_tv = get_inner_expanded_tv<5>(input2.desc);
+    tensor_view_t<5> T_tv  = get_inner_expanded_tv<5>(target.desc);
+    tensor_view_t<5> O_tv  = get_inner_expanded_tv<5>(output.desc);
+    uint64_t tensor_size   = target.desc.GetElementSize();
+    double sum_loss        = 0;
+
+    for(uint64_t gid = 0; gid < tensor_size; ++gid)
```

Comment:
Follow [my script](https://github.com/moreh-dev/MIOpen/blob/0049710197fe76cea9c35ec70dbec59ad08b2ee0/test/cpu_smoothl1loss.hpp#L33) for shorten and parallelism code.

Reply:
resolved